assets/src/scripts/editoria11y-tinymce.js

```diff
@@ -0,0 +1,298 @@
+						exp2.text = EXPORT_SHIM;
+						iframeWin.document.head.appendChild( exp2 );
+					}
+				} catch ( e ) { /* capture constructor polling error */ }
+				if ( iframeWin.Ed11y ) {
+					return done();
+				}
+				if ( window.Ed11y && ! iframeWin.Ed11y ) {
+					iframeWin.Ed11y = window.Ed11y;
+					return done();
+				}
+				if ( loops++ > 40 ) {
+					return done();
+				}
+				setTimeout( waitCtor, POLL_INTERVAL );
+			} )();
+			return;
+		}
+		const s = iframeWin.document.createElement( 'script' );
+		s.src = parentScript.src;
+		/**
+		 *
+		 */
+		s.onload = function () {
+			let loops2 = 0;
+			( function confirmCtor() {
+				try {
+					if ( ! iframeWin.Ed11y ) {
+						const exp3 = iframeWin.document.createElement( 'script' );
+						exp3.text = EXPORT_SHIM;
+						iframeWin.document.head.appendChild( exp3 );
+					}
+				} catch ( e2 ) { /* ignore post-load export error */ }
+				if ( iframeWin.Ed11y ) {
+					return done();
+				}
+				if ( window.Ed11y && ! iframeWin.Ed11y ) {
+					iframeWin.Ed11y = window.Ed11y;
+					return done();
+				}
+				if ( loops2++ > 25 ) {
+					return done();
+				}
+				setTimeout( confirmCtor, 120 );
+			} )();
+		};
+		/**
+		 *
+		 */
+		s.onerror = function () {
+			done();
+		};
+		iframeWin.document.head.appendChild( s );
+	}
+
+	/**
+	 * Ensure stylesheet is present in iframe.
+	 *
+	 * @param {Document} iframeDoc
+	 */
+	function ensureIframeStyles( iframeDoc ) {
+		const parentCss = document.querySelector( 'link[href*="editoria11y"][href$=".css"], link[href*="editoria11y.min.css"]' );
+		if ( parentCss && ! iframeDoc.querySelector( 'link[href="' + parentCss.href + '"]' ) ) {
+			const l = iframeDoc.createElement( 'link' );
+			l.rel = 'stylesheet';
+			l.href = parentCss.href;
+			iframeDoc.head.appendChild( l );
+		}
+	}
+
+	/**
+	 * Create Ed11y instance for an editor iframe.
+	 *
+	 * @param {object} editor
+	 */
+	function initEd11yInstance( editor ) {
+		const iframe = editor.iframeElement || document.getElementById( editor.id + '_ifr' );
+		if ( ! iframe || ! iframe.contentWindow || ! iframe.contentDocument ) {
+			return;
+		}
+		const w = iframe.contentWindow;
+		const d = w.document;
+		if ( ! d.body ) {
+			return setTimeout( function () {
+				initEd11yInstance( editor );
+			}, POLL_INTERVAL );
+		}
+		if ( ! /\btinymce\b/.test( d.body.id ) ) {
+			d.body.id = 'tinymce';
+		}
+		ensureIframeStyles( d );
+		ensureEd11yInIframe( w, function () {
+			if ( ! w.Ed11y ) {
+				return;
+			}
+			if ( w.__pbEd11yInstance ) {
+				return;
+			}
+			const opts = buildEd11yOptions();
+			try {
+				w.__pbEd11yInstance = new w.Ed11y( opts );
+				w.__pbEd11yLastFull = Date.now();
+				w.__pbEd11yVisible = true;
+			} catch ( e ) { /* ignore instantiation error */ }
+		} );
+	}
+
+	/**
+	 * Run full or incremental recheck.
+	 *
+	 * @param {object} editor
+	 * @param {boolean} forceFull
+	 */
+	function runEd11yRecheck( editor, forceFull ) {
```

Comment:
Track last full scan timestamp. Do checkAll() every 12s (or forced) else attempt incrementalCheck(); fallback to checkAll() if incremental not present

Reply:
After a conversation with John he suggested I take a look at `editableContents` parameter (see [here](https://github.com/pressbooks/pressbooks/pull/4177)) to allow Ed11y to watch for changes.